### PR TITLE
[dong] improvement - show loading indicator (small/big) when sync transaction (after create new transaction - like donate) #556

### PR DESCRIPTION
**Pull request by phongvu2812**
# Ticket
#556

# General
- [ ] No unnecessary files, debug logs, or commented code included.
- [ ] PR has at least 1 peer approval before senior review.
- [ ] I fully understand the code I submitted.
- [ ] No hardcoded configuration or secrets.
- [ ] **Environment Variables** changes? (Leave unchecked if N/A)

# Design & Solution Clarity
- [ ] A design diagram (sequence / flow) is attached or linked.
- [ ] The implementation matches the design.
- [ ] All major flows and decision points are represented in the diagram.
- [ ] Error handling and edge cases are clearly defined in the design.

Design link (diagram / doc):
`<paste link here>`

# Testing
- [ ] Unit tests cover main logic and important edge cases.

> Note: Complex logic must not rely on manual testing only.

# UI evidence (For frontend only)
`<paste screenshots here>`

# Peer Review Checklist
- [ ] I understand the overall solution and flow.
- [ ] The code follows the provided design diagram.
- [ ] No obvious logic or security issues found.
- [ ] Tests are sufficient for the scope of this change.

### CODE REVIEW

File: indexer/internal/storage/postgres.go [hunk 1]
```diff
@@ -854,6 +854,44 @@ func (p *PostgresConnector) insertBlockAndTransactions(ctx context.Context, bloc
...
+
+			for addr := range affectedDonationAddrs {
+				wg.Add(1)
+				sem <- struct{}{}
+
+				go func(toAddress string) {
+					defer wg.Done()
+					defer func() { <-sem }()
+
+					if err := services.SendSocketEventDirect(toAddress, "DONATION_RECEIVED", nil); err != nil {
```
Comment (vietdangquoc) [inline indexer/internal/storage/postgres.go]: DONATION_RECEIVED const

**Review by vietdangquoc — APPROVED** [review]

**Review by vietdangquoc — APPROVED** [review]

File: frontend/modules/donation-campaign/components/campaign-detail/shared/campaign-info/campaign-header.tsx [hunk 2]
```diff
@@ -13,10 +13,40 @@ import { useState } from 'react';
...
 export function CampaignHeader({ campaign }: { campaign: DonationCampaign }) {
   const [currentCampaign, setCurrentCampaign] = useState(campaign);
   const { user } = useUser();
+  const { mutate: refreshRaised } = useRefreshCampaignRaised({
+    onSuccess: (data) => {
+      handleRefreshData(data.total_amount, data.current_balance, data.total_withdrawn);
+    },
+  });
+
+  const { wsManager } = useJoinRoom(campaign.donation_wallet, true);
```
Comment (hoangdohuy) [inline frontend/modules/donation-campaign/components/campaign-detail/shared/campaign-info/campaign-header.tsx:35]: handle thêm refresh cả thông tin feed + activities ở dưới

File: indexer/internal/storage/postgres.go [hunk 3]
```diff
@@ -910,6 +910,44 @@ func (p *PostgresConnector) insertBlockAndTransactions(ctx context.Context, bloc
...
+			if txObj.TransactionExtraInfoType == common.TransactionExtraInfoDonationCampaign &&
+				txObj.Status != nil && (*txObj.Status == (uint64)(pb.TransactionStatus_CONFIRMED) || *txObj.Status == (uint64)(pb.TransactionStatus_FINALIZED)) &&
+				txObj.ToAddress != "" {
+				affectedDonationAddrs[txObj.ToAddress] = struct{}{}
+			}
+		}
+
+		if len(affectedDonationAddrs) > 0 {
+			maxConcurrency := 50
+			sem := make(chan struct{}, maxConcurrency)
```
Comment (hoangdohuy) [inline indexer/internal/storage/postgres.go]: hiện tại không cần implement semaphore ở đây đâu.
Sau này sẽ phải optimize ở trong services (thằng service ở câu services.SendSocketEventDirect) sau
